### DISCUSSION TITLE
[iOS] StageVideo doesn't playback audio from MP4 H.264 videos

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#2933] [Q&A]
On iPhone 11 (iOS 17.1.1) there's no sound playback with StageVideo. 
When testing on iPad AIR (iOS 12.5.5 and 12.5.7) sound works fine.

Is it some hardware restriction or am I missing something?

**Comment by D5CN** [comment]
it's because the License of mp4 format was outdate.now air is not support mp4 format.
you can transform your mp4 file to flv,and try again

**Reply by Jason0247** [reply]
Do you still have tools to convert to FLV format?

**Reply by D5CN** [reply]
what platform your software work?PC
i writed a code that can call ffmpeg to covert mp4 to flv. but just can run in PC

**Reply by Jason0247** [reply]
I am using Windows

**Reply by Jason0247** [reply]
Can't air SDK play MP4 videos on iOS now? Which version did you start with?

**Comment by Jason0247** [comment]
Have you tried transcoding MP4 videos? Convert to H.264?
I have encountered situations where there is only sound but no image before, and it works normally after transcoding. @MalacTheLittle

**Reply by itlancer** [reply]
Yes, try after transcoding. May be inside your MP4 used high profiles https://github.com/airsdk/Adobe-Runtime-Support/issues/139 or it MP4V. Such formats not supported by AIR.

**Reply by MalacTheLittle** [reply]
Since my knowledge on video formats and codecs is very basic, I lean on Adobe Media Encoder for transcoding. I tried different settings, none resulted with sound playback on iPhone 11. 
Can you recomend some different video transcoding app?

**Reply by Jason0247** [reply]
You can use a format factory

**Comment by ajwfrost** [comment]
Hi @MalacTheLittle  - we just checked this and have a stage video working fine with audio on an iPhone 11, iOS 17.1.2. So it may be down to the video and the specific codecs being used here... 
Would it be possible for you to upload the mp4 file using the below link so that we can try with this and see if we can find the problem?
https://transfer.harman.com/requests/pT48BGyEnUaqmo5mQkdADe

thanks

**Reply by ajwfrost** [reply]
Thanks for the file so quickly! .. but that also worked on the handset here.

Are you able to check using this simple test case:

```
package {
  import flash.display.*;
  import flash.events.*;
  import flash.geom.*;
  import flash.media.*;
  import flash.net.*;

  public class Main extends Sprite {

    private var _sv : StageVideo;
    private var _nc : NetConnection;
    private var _ns : NetStream;

    public function Main()
    {
      trace("HARMAN: starting up");
      stage.addEventListener(StageVideoAvailabilityEvent.STAGE_VIDEO_AVAILABILITY, onSVAvailable);
    }

    private function onSVAvailable(e:StageVideoAvailabilityEvent) : void
    {
      trace("HARMAN: " + e.toString());
      if (e.availability == StageVideoAvailability.AVAILABLE)
      {
        trace("Starting video");
        _sv = stage.stageVideos[0];
        _sv.viewPort = new Rectangle(0, 0, stage.stageWidth, stage.stageHeight);
        _nc = new NetConnection();
        _nc.addEventListener(NetStatusEvent.NET_STATUS, onNetStatus);
        _nc.connect(null);
      }
    }

    private function onNetStatus(e:NetStatusEvent) : void
    {
      trace("HARMAN: Net Status " + e.info.code);
      switch (e.info.code)
      {
        case "NetConnection.Connect.Success":
          _ns = new NetStream(_nc);
          _ns.addEventListener(NetStatusEvent.NET_STATUS, onNetStatus);
          _ns.client = this;
          _sv.attachNetStream(_ns);
          _ns.play("v1.mp4");
          break;
      }
    }

    public function onMetaData(info:Object) : void
    {
    }
  }
}
```

**Reply by MalacTheLittle** [reply]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/56675407/408c7acb-37cf-44a6-aaa4-2d2bde7cd5a1)

OK, I don't know if I would laugh or cry, but the problem was this small switch LOL
And the reason why that didn't cross my mind is that .mp3 files are reproduced normaly regardless of that switch. Also, If I run youtube video in Safari, sound also reproduces regardless of that switch.

Very odd hardware feature...

**Comment by MalacTheLittle** [comment]
Hardware switch for Silent mode was turned on so StageVideo wasn't reproducing sound.
Solution: turn Silent mode switch off.

Thank you all for help

**Reply by ajwfrost** [reply]
:-) we've all been there!
